src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -155,7 +155,7 @@ public static void markDirty()
      *
      * @param id the colonies id.
      */
-    public static void deleteColony(final int id)
+    public static void deleteColony(final int id, boolean canDestroy)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Provide multiple methods instead of using "canDestroy" to determine which action to take. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2301)